snafu-derive/src/lib.rs

```diff
@@ -1121,7 +1184,9 @@ impl<'a> ErrorCompatImpl<'a> {
 
 impl<'a> quote::ToTokens for ErrorCompatImpl<'a> {
     fn to_tokens(&self, stream: &mut proc_macro2::TokenStream) {
-        let original_generics = &self.0.generics;
+        let original_generic_types = &self.0.provided_generic_types_without_defaults();
+        let original_lifetimes = &self.0.provided_generic_lifetimes();
+        let original_generics = original_lifetimes.into_iter().chain(original_generic_types);
```

Comment:
Since we repeat this pattern 3 times, perhaps it's worth creating a helper method:

```rust
let original_generics = self.0.provided_some_reasonable_name_here();
```